### DISCUSSION TITLE
MySQL Options field resets itself to text after fetching tables

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#4080] [Help] [answered]
Hi Budibase!

I was working on an application and encountered the following problem: the options in the 'options' fields were reset without warning.
I don't have the details because I didn't pay close attention, here are the steps:
- I have MySQL tables that have 'Options' fields with values to select.
- I fetch the tables from MySQL to add a new table (or something else)
- All the options fields are now set to 'Text' and no longer contain any of the saved choices.
- The choices are lost, the field type is changed and therefore the application does not work (form dropdowns are empty)

If this is the intended behaviour (I have a doubt about this), could budibase at least provide a warning?
I think Budibase should keep the type and values related to the field, I will try to reproduce the bug more accurately if necessary.

Thanks for the help and advice on this problem!

**Comment by mike12345567** [comment, marked as answer]
Hi @diogenesbrussels - can I check what version of Budibase you're currently using? This issue should be fixed in the latest release (a few days ago) - https://github.com/Budibase/budibase/pull/4019

**Reply by diogenesbrussels** [reply]
Hi @mike12345567,

Thank you for the quick answer, I didn't see #4019 with my quick search.
I was running 1.0.26, I updated to 1.0.42 this morning.

I will test this again in a few minutes & let you know!

**Reply by diogenesbrussels** [reply]
Great, it works as expected!

**Comment by diogenesbrussels** [comment]
Hi @mike12345567,

I had very similar bug with a Multi-select column, after fetching table again the column type is set to json by budibase and all pre-filled values are lost.
I can't switch the column type back to Multi-select after this.
I am using v1.0.46.

Thank you!

**Reply by mike12345567** [reply]
Hi @diogenesbrussels - two questions about this:

1. Did you add the multi-select column to your MySQL table through Budibase?
2. What is the actual type of the column in MySQL?

**Reply by diogenesbrussels** [reply]
1. Yes
2. json

**Reply by mike12345567** [reply]
Ah thanks @diogenesbrussels - I'll raise a bug around this, we use a JSON array for the multi-select when using MySQL, the new JSON type that is being added overlaps with this - I'll update it so that you can select between JSON and Multi-select types.

EDIT: issue created: https://github.com/Budibase/budibase/issues/4192

**Comment by Villevalld** [comment]
Hi folks,

I have similar issue using mysql db.
After refetching the tables all the "Text" fields are switched to "Long Form Text".
Is there any chance to switch it back?
MySQL field remains as TEXT.
Budibase version is 1.0.90

Thanks
Denis
